accesscontrol/jwk.go

```diff
@@ -57,20 +65,21 @@ func (j *JWK) UnmarshalJSON(data []byte) error {
 	}
 
 	var key interface{}
+	jwk := JWK{KeyID: raw.Kid, Algorithm: raw.Alg, Use: raw.Use}
 
 	switch raw.Kty {
 	case "RSA":
-		if len(raw.X5c) > 0 {
-			certificate, err := x509.ParseCertificate(raw.X5c[0].data)
-			if err != nil {
-				// TODO log warning properly
-				fmt.Printf("Invalid x5c: %v\n", err)
-				return nil
-			}
+		key, err = getPublicKeyFromX5c(raw.X5c)
+		if err != nil {
+			// TODO log warning properly
```

Comment:
In general this Unmarshal method should return all errors, especially any "raw" unmarshal work, if this is broken we should log this as error. The ignored ones could being wrappend into an own error type which should be type checked on the callee side for logging purposes (containing multiple ones).
To inject logging, I suggest to pass the `logrus.Entry` from `configureJWKS` to `ac.NewJWKS` and handle them by type during `Load`.

Additionally the JWKS method Receiver names are different (`j` vs `jwks`). 
Additionally the JWK method Receiver should be all pointer ones. (`jwk.go:41`)

Reply:
/cc @alex-schneider

Reply:
> solve some code todos […]  and handle them by type during Load.
I disagree: If we still intend to change the underlying lib this will be unnecessary work. The idea was to keep this task small.

>  Additionally the JWK method Receiver should be all pointer ones. (jwk.go:41)
I guess we can drop `func (j JWK) MarshalJSON() ([]byte, error)` compeletely as no one uses it.

Reply:
> 

ok. Let's see how this will look like with the lib update.
@alex-schneider pls ignore this todo for now.

Reply:
> Additionally the JWKS method Receiver names are different (j vs jwks).
I'll change this in my branch (JWKS for OIDC).